graphql/lessonplan/mutations.ts

```diff
@@ -0,0 +1,139 @@
+import { MaxLength, ArrayMaxSize } from 'class-validator';
+import { Role } from '../authorizations';
+import { GraphQLContext } from '../context';
+import { getLogger } from '../../common/logger/logger';
+import { generateLessonPlan } from '../../common/lessonplan/generate';
+import { course_subject_enum, pupil_state_enum, school_schooltype_enum } from '@prisma/client';
+import { PrerequisiteError, ClientError } from '../../common/util/error';
+import { getStateFullName } from '../../common/util/stateMappings';
+
+const logger = getLogger(`LessonPlan Mutations`);
+
+// Maximum prompt length (in characters)
+const MAX_PROMPT_LENGTH = 10000;
+
+// Maximum number of files
+const MAX_FILES = 10;
+
+// Create a new enum for expected output fields
+enum LessonPlanField {
+    TITLE = 'title',
+    LEARNING_GOAL = 'learningGoal',
+    AGENDA_EXERCISES = 'agendaExercises',
+    ASSESSMENT = 'assessment',
+    HOMEWORK = 'homework',
+    RESOURCES = 'resources',
+}
+
+// Register the LessonPlanField enum
+registerEnumType(LessonPlanField, {
+    name: 'LessonPlanField',
+    description: 'Fields that can be generated for a lesson plan',
+});
+
+@InputType()
+class GenerateLessonPlanInput {
+    @Field(() => [String], { nullable: true })
+    @ArrayMaxSize(MAX_FILES, { message: `Number of files must not exceed ${MAX_FILES}` })
+    fileUuids?: string[];
+
+    @Field(() => course_subject_enum)
+    subject: course_subject_enum;
+
+    @Field(() => Int)
+    grade: number;
+
+    @Field(() => Int)
+    duration: number;
+
+    @Field(() => pupil_state_enum)
+    state: pupil_state_enum;
+
+    @Field(() => String)
+    @MaxLength(MAX_PROMPT_LENGTH, { message: `Prompt must not exceed ${MAX_PROMPT_LENGTH} characters` })
+    prompt: string;
+
+    @Field(() => [LessonPlanField], { nullable: true })
+    expectedOutputs?: LessonPlanField[];
+
+    @Field(() => school_schooltype_enum)
+    schoolType: school_schooltype_enum;
+
+    @Field(() => String, { nullable: true, defaultValue: 'German' })
+    language?: string;
+}
+
+@ObjectType()
+class LessonPlanOutput {
+    @Field(() => String, { nullable: true })
+    title?: string;
+
+    @Field(() => course_subject_enum)
+    subject: course_subject_enum;
+
+    @Field(() => String)
+    grade: string;
+
+    @Field(() => Int)
+    duration: number;
+
+    @Field(() => String, { nullable: true })
+    learningGoal?: string;
+
+    @Field(() => String, { nullable: true })
+    agendaExercises?: string;
+
+    @Field(() => String, { nullable: true })
+    assessment?: string;
+
+    @Field(() => String, { nullable: true })
+    homework?: string;
+
+    @Field(() => String, { nullable: true })
+    resources?: string;
+}
+
+@Resolver()
+export class MutateLessonPlanResolver {
+    @Mutation(() => LessonPlanOutput)
+    @Authorized(Role.INSTRUCTOR, Role.ADMIN)
+    async generateLessonPlan(@Ctx() context: GraphQLContext, @Arg('data') data: GenerateLessonPlanInput): Promise<LessonPlanOutput> {
```

Comment:
Naming convention would be <subject><verb>, so `lessonPlanGenerate` - but thats also inconsistent in other places anyways ...